java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java

```diff
@@ -202,22 +221,48 @@ public static URI urlToFSPath(String urlString, String name) {
             host = url.getHost();
             path = url.getPath();
 
+            // Case 1
             if ("localhost".equals(host)) {
                 return url.toURI();
             }
+            String qPath = Pattern.compile("&")
```

Comment:
not sure how much more overhead this pattern is but i would just use `Arrays.stream(url.getQuery().split("&"))` for something simple like this.